libs/storage/Tsavorite/cs/src/core/Index/Tsavorite/Implementation/Revivification/RevivificationManager.cs

```diff
@@ -13,20 +13,17 @@ internal struct RevivificationManager<Key, Value>
         internal RevivificationStats stats = new();
 
         internal readonly bool IsEnabled = false;
-        internal readonly int FixedValueLength;
+        internal readonly int FixedValueLength => Unsafe.SizeOf<Value>();
         internal bool restoreDeletedRecordsIfBinIsFull;
         internal bool useFreeRecordPoolForCTT;
 
-        internal readonly bool IsFixedLength => FixedValueLength != 0;
+        internal readonly bool IsFixedLength { get; }
```

Comment:
Is this change needed? If original implementation would be more efficient.

Reply:
The revivifaction logic was broken due to how isFixedLen was flowed in to this ctor. This was quickest fix I could come up to restore previous logic. Shouldn't JIT be able to consider this bool as constant in next tier as it is `readonly`?

Reply:
I wrecked my brain and couldn't come up how one of these variations would result less branching or worse CQ.. What am I missing?

Reply:
I agree. I think what you have is fine after taking a closer look.

Reply:
@TedHartMS - pls check this thread as well.

Reply:
This change fixes 9bdb889 that set FixedValueLength always, rather than setting it only if isFixedLen is true. While the fix in ef5a01a was sufficient to fix the GC hole, the change in 9bdb889 made it consistent with the pattern of calling Unsafe.SizeOf only in properties. This requires a separate bool value for IsFixedLength. Unfortunately, as you've discovered, this bool can't be considered a constant even though it is readonly.